src/url-setters.cpp

```diff
@@ -203,4 +203,20 @@ namespace ada {
     }
   }
 
+  void url::set_href(const std::string_view input) {
+    ada::url out = ada::parse(input);
+
+    if (out.is_valid) {
+      set_protocol(out.get_protocol());
+      set_username(out.get_username());
+      set_password(out.get_password());
+      set_host(out.get_host());
+      set_hostname(out.get_hostname());
+      set_port(out.get_port());
+      set_pathname(out.get_pathname());
+      set_hash(out.get_hash());
+      set_search(out.get_search());
+    }
```

Comment:
See https://github.com/ada-url/ada/pull/161

I am proposing that some of these setters return a bool on success. If so, then you should stop after the first setter that failed and report failure.

Whatever you do, on failure, the URL could be left in a bad state so if failure is possible... we need to be a bit careful.

Reply:
For this particular example, there's no need to check if each iteration is succeeded or not, since we check it on `out.is_valid` and this `if` statement should override each values. If the implementation is correct, we should never get a scenario where we have to check if each setter succeeded or not.